pkg/bootstrap/bootstrap.go

```diff
@@ -370,10 +371,16 @@ func rewriteChart(fileName string, info os.FileInfo, systemDefaultRegistry strin
 
 	// Generally we should avoid using Set on HelmCharts since it cannot be overridden by HelmChartConfig,
 	// but in this case we need to do it in order to avoid potentially mangling the ValuesContent field by
-	// blindly appending content to it in order to set the global.systemDefaultRegistry value.
+	// blindly appending content to it in order to set values.
 	if chart.Spec.Set == nil {
 		chart.Spec.Set = map[string]intstr.IntOrString{}
 	}
+
+	if chart.Spec.Set["global.rke2DataDir"].StrVal != dataDir {
```

Comment:
What's the behavior here if the objects at that key doesn't exist and we try to call `Strval` on it?

Reply:
`chart.Spec.Set` is typed as `map[string]intstr.IntOrString{}`, so if that's empty the map gives you back a default IntOrString struct. StrVal on the default struct is just an empty string.

Reply:
Cool. Just wanted to make sure we weren't introducing a point of panic.